### PR TITLE
docs(HMS-2680): add contributing guidelines

### PR DESCRIPTION
**Pull request by avisiedo**
Add a contributing guidelines aligned with the rest of the repositories.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -36,27 +36,6 @@ $(NODE_BIN)/%: package.json package-lock.json
 	npm install
 	touch $(NODE_BIN)/*
 
-$(OAPI_CODEGEN):
```
Comment (frasertweedale) [inline Makefile:39]: Was this removal intended?  CI is failing due to missing `oapi-codegen` target.

Comment (avisiedo) [reply Makefile:39]: Thanks! Yes it is intentional; I miss the pipeline. :+1:

For the idmsvc-api repository we don't need to generate the code, as it is part of the idmsvc-backend.

IMHO linters and openapi validation should be enough on this repository.

I will remove the remaining of this in the pipeline.

**Comment by frasertweedale** [comment]
Conflicts now :/

**Comment by avisiedo** [comment]
@frasertweedale  Thanks! I forgot to fetch repository changes on a long time :+1: now is rebased on top of `main`

**Review by frasertweedale — APPROVED** [review]

**Pull request merged**